store-implementation/parquet-store/src/main/java/uk/gov/gchq/gaffer/parquetstore/operation/addelements/impl/AggregateAndSortTempData.java

```diff
@@ -56,25 +68,50 @@ public AggregateAndSortTempData(final ParquetStore store, final SparkSession spa
             currentDataDir = null;
         }
         for (final String group : schemaUtils.getEdgeGroups()) {
-            final String currentDataInThisGroupDir;
-            if (currentDataDir != null) {
-                currentDataInThisGroupDir = ParquetStore.getGroupDirectory(group, ParquetStoreConstants.VERTEX, currentDataDir);
-            } else {
-                currentDataInThisGroupDir = null;
+            if (groupToSplitPoints.containsKey(group)) {
+                final String currentDataInThisGroupDir;
+                if (currentDataDir != null) {
+                    currentDataInThisGroupDir = ParquetStore.getGroupDirectory(group, ParquetStoreConstants.VERTEX, currentDataDir);
+                } else {
+                    currentDataInThisGroupDir = null;
+                }
+                final Collection<Integer> splits = groupToSplitPoints.get(group).values();
+                for (final int i : splits) {
+                    final Set<String> currentGraphFiles = new HashSet<>();
+                    if (currentDataInThisGroupDir != null) {
+                        currentGraphFiles.add(currentDataInThisGroupDir + "/part-" + zeroPad(String.valueOf(i), 5) + "*.parquet");
+                    }
+                    tasks.add(new AggregateGroupSplit(group, ParquetStoreConstants.SOURCE, store, currentGraphFiles, spark, i));
+                }
             }
-            tasks.add(new AggregateAndSortGroup(group, ParquetStoreConstants.SOURCE, store, currentDataInThisGroupDir, spark));
         }
         for (final String group : schemaUtils.getEntityGroups()) {
-            final String currentDataInThisGroupDir;
-            if (currentDataDir != null) {
-                currentDataInThisGroupDir = ParquetStore.getGroupDirectory(group, ParquetStoreConstants.SOURCE, currentDataDir);
-            } else {
-                currentDataInThisGroupDir = null;
+            if (groupToSplitPoints.containsKey(group)) {
+                final String currentDataInThisGroupDir;
```

Comment:
The code inside this 'if' statement is almost identical to the code in the 'if' statement above - can you refactor it into a separate method and re-use?